### PR TITLE
vault: derive token using `create_from_role`

### PR DESCRIPTION
**Pull request by lgfa29**
Fallback to the ACL role defined in the client's `create_from_role` configuration when using the JWT flow and the task does not specify a role to use.

### CODE REVIEW

File: client/config/config.go [hunk 1]
```diff
@@ -179,6 +179,8 @@ type Config struct {
 	ConsulConfigs map[string]*structsc.ConsulConfig
 
 	// VaultConfig is this Agent's default Vault configuration
+	//
+	// Deprecated: use GetVaultConfigs() instead.
 	VaultConfig *structsc.VaultConfig
```
Comment (lgfa29) [inline client/config/config.go:184]: I keep forgetting that I'm not supposed to read from this value. Wit this, my editor screams at me if I try to 😅

**Comment by lgfa29** [comment]
Ops,. broke stuff 😬

File: client/config/config.go (hunk 1 above)
Comment (tgross) [reply client/config/config.go:184]: You and me both. After we've shipped the beta and have a moment to breathe, I'd like to see if we could remove it entirely. It was too gnarly of a refactoring to do while we were also trying to ship the feature :grinning:

**Review by tgross — APPROVED** [review]
LGTM

**Pull request merged**